django/db/models/deletion.py

```diff
@@ -64,7 +64,7 @@ class Collector(object):
     def __init__(self, using):
         self.using = using
         # Initially, {model: {instances}}, later values become lists.
-        self.data = {}
+        self.data = OrderedDict()
```

Comment:
I wonder if this affect performance a lot. Do you think you could run [djangobench](https://github.com/django/djangobench) against this change?